### ISSUE TITLE
TextFloat does not work under Wayland

### ISSUE THREAD

**Issue by michaelgregorius** [LMMS/lmms#7626]
### System Information

Arch Linux with Wayland

### LMMS Version(s)

master

### Bug Summary

Positioning the text float window does not work under Wayland. All text floats are shown at position (0, 0) instead of at the lower left corner of the main window like under X11.

The reason is that `TextFloat::moveGlobal` uses [QWidget::move](https://doc.qt.io/qt-6/qwidget.html#pos-prop) which is not supported under Wayland.

### Expected Behaviour

The text float should not show at (0, 0).

Proposal: Implement a dedicated area where messages are shown and collected instead of using an ephemeral text float which users might miss to notice anyway.

### Steps To Reproduce

Open a project that has been created with an older version of LMMS so that the text float which shows that information is displayed.

### Please search the issue tracker for existing bug reports before submitting your own.

- [X] I have searched all existing issues and confirmed that this is not a duplicate.

**Comment by regulus79** [comment]
I am on Arch Linux, Wayland, GNOME 47, and I cannot reproduce this issue. The `TextFloat`s appear at the bottom left as usual for me.

**Comment by michaelgregorius** [comment]
Here's how it looks under KDE/Plasma with Wayland:

[TextFloatWithKDEPlasma.webm](https://github.com/user-attachments/assets/a5b5a46f-8ec8-4998-a522-9a904f4a6913)

It's a bit surprising that it seems to work under Gnome as it is stated that Wayland does not allow clients to place windows. Are you sure that you are triggering a case where the text float window is a top-level window without a parent, e.g. in the way as described above?

**Comment by regulus79** [comment]
https://github.com/user-attachments/assets/4cf35a83-05d2-4cd8-9a34-af2a9993255f

**Comment by Rossmaxx** [comment]
So the glitch seen in my video might be a one-off thing which is unique to vsts.

**Comment by Rossmaxx** [comment]
Oops, wrong issue. Thought this was the PR.

**Comment by regulus79** [comment]
On closer inspection it appears that LMMS may not have been running on Wayland. Although GNOME says that my windowing system is Wayland, and everything I have checked appears to show that I am running Wayland, LMMS gives the following warning when it starts up: 
`Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.`
I am not an expert on windowing systems, so I'm not sure what else it would be running on (XWayland?) But sure enough, when I set `QT_QPA_PLATFORM=wayland`, the LMMS window looks different, and I can reproduce this issue of the text floats spawning in the top left.

**Comment by michaelgregorius** [comment]
Yes, I guess that it is running through XWayland and that it therefore shows the same behavior as under X11. Although it's also interesting that the placing of windows seems to be allowed then under this setup. However that works. :sweat_smile:

**Comment by tresf** [comment]
> Yes, I guess that it is running through XWayland and that it therefore shows the same behavior as under X11. Although it's also interesting that the placing of windows seems to be allowed then under this setup. However that works. 😅

As an additional details, we've only ever shipped the xcb qt lib with our AppImage.  We should be able to change this relatively easily, probably worthy of just a PR that references this issue.

**Comment by allejok96** [comment]
I sometimes find it annoying that the big TextFloats remain when LMMS is minimized. Especially if LMMS locks up or during debugging. Would it be possible to make TextFloat a child of the main window and call raise() to draw it on top of all other widgets instead of making it a separate window?

Does the small tooltips for knobs work on Wayland?

**Comment by allejok96** [comment]
I sometimes find it annoying that the big TextFloats remain when LMMS is minimized. Especially if LMMS locks up or during debugging. Would it be possible to make TextFloat a child of the main window and call raise() to draw it on top of all other widgets instead of making it a separate window?

Does the small tooltips for knobs work on Wayland?

**Comment by michaelgregorius** [comment]
> Does the small tooltips for knobs work on Wayland?

Yes, it does.

**Comment by messmerd** [comment]
This appears to be a Qt bug: https://qt-project.atlassian.net/browse/QTBUG-85297

It should be fixed once we upgrade the Linux builds to at least Qt 6.4.

**Comment by michaelgregorius** [comment]
Thanks for checking @messmerd!

I have tested by building LMMS against Qt 6.10.2 and I can confirm that the bug is fixed there.

**Comment by messmerd** [comment]
Let's keep this open until the Linux builds are upgraded to use Qt 6.4+. They're currently on Qt 5.15, so the bug is still present.
